src/main/java/org/opensearch/ad/util/IndexUtils.java

```diff
@@ -116,17 +126,27 @@ public String getIndexHealthStatus(String indexOrAliasName) throws IllegalArgume
      * @return The number of documents in an index. 0 is returned if the index does not exist. -1 is returned if the
      * request fails.
      */
-    /*  @anomaly-detection - commented until we have support for SDKRestClient.stats() https://github.com/opensearch-project/opensearch-sdk-java/issues/620
-    @Deprecated
     public Long getNumberOfDocumentsInIndex(String indexName) {
         if (!clusterService.state().getRoutingTable().hasIndex(indexName)) {
             return 0L;
         }
-        IndicesStatsRequest indicesStatsRequest = new IndicesStatsRequest();
-        Optional<IndicesStatsResponse> response = clientUtil.timedRequest(indicesStatsRequest, logger, client.admin().indices()::stats);
-        return response.map(r -> r.getIndex(indexName).getPrimaries().docs.getCount()).orElse(-1L);
+        IndicesStatsRequest indicesStatsRequest = new IndicesStatsRequest.Builder().build();
+
+        CompletableFuture<IndicesStatsResponse> indicesStatsFutureResponse = openSearchAsyncClient
+            ._transport()
+            .performRequestAsync(indicesStatsRequest, IndicesStatsRequest._ENDPOINT, TransportOptions.builder().build());
+        // Optional<IndicesStatsResponse> response = clientUtil.timedRequest(indicesStatsRequest, logger, client.admin().indices()::stats);
+
+        IndicesStatsResponse indicesStatsResponse;
+        Long numberOfDocumentsInIndex = -1L;
```

Comment:
Why used a boxed `long` here?  I see the method return type is a `Long` but that's normally used if you want to return `null` (and may have originally had that logic before JDK8 and `Optional`).  And perhaps that's why this method is now deprecated.  Right now we're returning either 0 or -1 in error conditions, so we can save the boxing until the final return.

Reply:
I have used a boxed long here because if I don't use this then I have to add an extra check here and then store the value of the count in some other variable.

`return indicesStatsResponse.indices().get(indexName).primaries().docs().count()!=null? indicesStatsResponse.indices().get(indexName).primaries().docs().count(): -1`

Reply:
Got it, was thinking of if this was a new method we shouldn't do this and if it's a deprecated method we should call the replacement, but this is really just migration work on a poorly written (and now deprecated) method :)